crates/assistant/src/assistant_settings.rs

```diff
@@ -12,14 +13,25 @@ pub enum OpenAiModel {
     Four,
     #[serde(rename = "gpt-4-1106-preview")]
     FourTurbo,
+    Custom,
 }
 
 impl OpenAiModel {
+    pub fn convert_enum(model_name: String) -> OpenAiModel {
```

Comment:
A name like `from_str` here would be more conventional. We should also take a string slice, as we don't need ownership of the string:

```suggestion
    pub fn from_str(model_name: &str) -> OpenAiModel {
```